### DISCUSSION TITLE
How to expose user's password last updated time information?

### DISCUSSION THREAD

**Discussion by BradNut** [keycloak/keycloak#22811] [Q&A]
I am looking to expose the user's password last updated time as some sort of custom attribute. What would be the best way to do this? Would it be a custom event listener SPI for user password update and is that even an event that is exposed?

Also, is there already a database field that contains this information given that there is a password expiry policy and I would assume there's some entry it utilizes for this?

I have searched for one but cannot find reference to where that would be stored. Is this field only populated when the policy is active on the realm and where would that exist?

**Comment by christian-2** [comment]
Look at `"createdDate"` for `"type" : "password"` in [`CredentialRepresentation`](https://www.keycloak.org/docs-api/22.0.1/rest-api/index.html#CredentialRepresentation), as returned by [`GET /admin/realms/{realm}/users/{id}/credentials`](https://www.keycloak.org/docs-api/22.0.1/rest-api/index.html#_users) (aka `kcadm.sh get -r $realm users/$id/credentials`).

**Comment by BradNut** [comment]
@christian-2 Yes that works! I'm not sure why I didn't see this before. Thank you!
